odoo/models.py

```diff
@@ -6206,6 +6206,20 @@ def filtered_domain(self, domain):
 
                 if comparator in ('like', 'ilike', '=like', '=ilike', 'not ilike', 'not like'):
                     value_esc = value.replace('_', '?').replace('%', '*').replace('[', '?')
+                    if comparator.endswith('ilike'):
+                        # ilike uses unaccent and lower-case comparison
+                        def unaccent_real(x):
+                            return self.pool.unaccent_python(x.lower()) if x else ''
+
+                        unaccent = unaccent_real
+                    else:
+                        def str_or_empty(x):
+                            return x or ''
+
+                        unaccent = str_or_empty
+                    value_esc = unaccent(value_esc)
+                    if not comparator.startswith('='):
+                        value_esc = '*' + value_esc + '*'
```

Comment:
Because it's called often in a loop so we save a test at each element.
As for consistency between '' and False, I've opened another PR: https://github.com/odoo/odoo/pull/148411